shell/platform/embedder/tests/embedder_unittests.cc

```diff
@@ -2562,6 +2562,36 @@ TEST_F(EmbedderTest, CanSendPointer) {
   message_latch.Wait();
 }
 
+TEST_F(EmbedderTest, RegisterChannelListener) {
+  auto& context = GetEmbedderContext(EmbedderTestContextType::kSoftwareContext);
+
+  fml::AutoResetWaitableEvent latch;
+  fml::AutoResetWaitableEvent latch2;
+  bool listening = false;
+  context.AddNativeCallback(
+      "SignalNativeTest",
+      CREATE_NATIVE_ENTRY([&](Dart_NativeArguments) { latch.Signal(); }));
+  context.SetChannelUpdateCallback([&](const FlutterChannelUpdate* update) {
+    if (strcmp(update->channel, "test/listen") == 0) {
```

Comment:
Can this be a `EXPECT_EQ` instead?

Also, could we add a `EXPECT_TRUE` on `update->listening`?